### PR TITLE
feat: use tsgo for build and type checks

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

To improve the performance for agentic development need to migrate to `tsgo`

**Description**

- Update `typescript` to latest transition version
- Add `tsgo` as dependency

### CODE REVIEW

**Comment by wemeetagain** [comment]
@lodekeeper can you look thru the files excluded for type checking that have been added here and create an issue documenting them and which functionalities need to be refactored to prevent importing things outside each package's rootDir? (We would like to re-include them once this refactoring is complete, so the issue documents the issue and the refactoring scope)

**Comment by lodekeeper** [comment]
@wemeetagain done — I created a tracking issue with the excluded file inventory + refactor scope + acceptance criteria: https://github.com/ChainSafe/lodestar/issues/8997\n\nIt documents each currently excluded path from PR #8992 and the concrete work needed to re-include them under tsgo/rootDir constraints.

File: packages/beacon-node/tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,24 @@
 {
   "extends": "../../tsconfig.json",
   "include": ["src", "test"],
+  "exclude": [
+    // Tests are intentionally excluded from the `check-types` compilation.
+    // TS6059 errors because those files fall outside this package's `rootDir`.
+    "test/utils/state-transition.ts",
+    "test/unit/chain/validation/attestation/validateAttestation.test.ts",
```
[the remark this answers is not in the document]
Comment (nazarhussain) [reply packages/beacon-node/tsconfig.json]: That's one edge case we can live with considering that `/test-utils` only meant to be used for testing.

**Comment by nazarhussain** [comment]
@codex @gemini-code-assist Please review this PR from fresh.

File: packages/api/src/beacon/routes/node.ts [hunk 2]
```diff
@@ -15,6 +15,10 @@ import {HttpStatusCode} from "../../utils/httpStatusCode.js";
 import {Endpoint, RouteDefinitions, Schema} from "../../utils/index.js";
 import {WireFormat} from "../../utils/wireFormat.js";
 
+// TODO: Workaround for tsgo import-elision bug: ensure this is treated as a runtime value.
+// https://github.com/microsoft/typescript-go/issues/2212
+void HttpStatusCode;
```
[the remark this answers is not in the document]
Comment (wemeetagain) [reply packages/api/src/beacon/routes/node.ts:20]: - https://github.com/ChainSafe/lodestar/issues/9022

**Review by wemeetagain — APPROVED** [review]
LGTM

Pretty minimal update to go to tsgo.
- import the new tsgo dependencies
- update vscode settings for proper vscode integration
- workaround for tsgo import-elison bug (tracked in #9022)
- move base tsconfig from tsconfig.build.json to tsconfig.common.json

**Review by nflaig — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.41.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.41.0) :tada:
